test/integration/options/require.spec.js

```diff
@@ -45,6 +46,51 @@ describe('--require', function() {
         /beforeAll[\s\S]+?beforeAll array 1[\s\S]+?beforeAll array 2[\s\S]+?beforeEach[\s\S]+?beforeEach array 1[\s\S]+?beforeEach array 2[\s\S]+?afterEach[\s\S]+?afterEach array 1[\s\S]+?afterEach array 2[\s\S]+?afterAll[\s\S]+?afterAll array 1[\s\S]+?afterAll array 2/
       );
     });
+
+    describe('support ESM when style=module or .mjs extension', function() {
+      before(function() {
+        if (!utils.supportsEsModules()) this.skip();
+      });
+
+      it('should run root hooks when provided via mochaHooks', function() {
+        return expect(
+          invokeMochaAsync(
+            [
+              '--require=' +
+                require.resolve(
+                  // as object
+                  '../fixtures/options/require/root-hook-defs-esm.fixture.mjs'
+                ),
+              '--require=' +
+                require.resolve(
+                  // as function
+                  '../fixtures/options/require/esm/root-hook-defs-esm.fixture.js'
+                ),
+              '--require=' +
+                require.resolve(
+                  // mixed with commonjs
+                  '../fixtures/options/require/root-hook-defs-a.fixture.js'
+                ),
+              require.resolve(
+                '../fixtures/options/require/root-hook-test.fixture.js'
+              )
+            ],
+            {
+              env: {
+                ...process.env,
```

Comment:
This is what I updated the parser options (parserOptions: 2018) in eslintrc for.

Considering mocha is documented with:
> As of v7.0.0, Mocha requires Node.js v8.0.0 or newer.
This seems safe enough for tests at least.

If there is an easier way to pass `--experimental-modules` to the mocha (instead of calling node directly) happy to use that instead...

Reply:
I apologize for the state of things (we're working on it; see #4293), but...

I wanted to get a subset of the integration tests (`test/integration/`) running in a browser, but haven't completed this work.  Stuff in `test/node-unit/` (and a few other places) is allowed to use ES2018 syntax, but literally everything _else_ under `test/` is limited to ES5.  It's probably OK to allow ES2018 in `test/integration/`, if we expect #4293 to get merged in the near future, since we'll be able to transpile everything down to where it needs to be.  I don't envision browser-based integration tests landing before then, so _IMO_ it's fine.  Others may disagree (@mochajs/core ?)

However, stuff in `test/unit/` and `test/browser` _cannot_, at this time, be ES2018, because these tests run in IE11.  So, please revert the change to `.eslintrc.yml`, then add `'test/integration/**/*.js'` to the list of overrides which already use `parserOptions: 2018` (L39 or so).

Reply:
Here, you're running `bin/mocha` (which is what `invokeMocha` does), but using the `NODE_OPTIONS` environment variable to set `--experimental-modules`. You don't need to do this; just pass `--experimental-modules` in your array of args.  `bin/mocha` knows what to do with it.

e.g.

```js
invokeMocha(['path/to/fixture', '--some-other-flag'].concat(+process.versions.node.split('.')[0] >= 13 ? [] : ['--experimental-modules'], function() {})
```

I realize this means you can just revert the change to `.eslintrc.yml` entirely, which is maybe what should happen.

Reply:
Reverted change to eslintrc

Didn't realize I could pass the flag directly though mocha! Since I don't need any lint changes, I don't think this is the place to reorganize what is allowed where.